dashboard/lib/expired_deleted_account_pii_scrubber.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require 'stringio'
+require 'cdo/aws/metrics'
+require 'cdo/aws/s3'
+require 'cdo/chat_client'
+require 'cdo/honeybadger'
+
+# Queries for accounts soft-deleted at least 28 days ago and scrub them
+# of PII (personally identifiable information).
+#
+# This renders the accounts unrecoverable but retains as much useful non-PII
+# data as possible.
+#
+# Logs activity to the user-accounts and cron-daily Slack channels, as well as Cloudwatch.
+class ExpiredDeletedAccountPiiScrubber
+  class SafetyConstraintViolation < RuntimeError; end
+
+  attr_reader :dry_run, :scrub_accounts_deleted_since, :max_accounts_to_scrub
+  alias :dry_run? :dry_run
+
+  LOGGING_NAMESPACE = 'Platform/PiiScrubber'
+  ACCOUNT_SCRUB_LIMIT = 8_000
+
+  def initialize(options = {})
```

Comment:
It is acceptable to use `**options` with documentation comments if the list is long or not fixed, but explicit keywords are preferred when the method's contract is well-defined:

```suggestion
  # @param dry_run [Boolean] ...
  # @param scrub_accounts_deleted_since [Time] ...
  # @param max_accounts_to_scrub [Integer] ...
  def initialize(dry_run: nil, scrub_accounts_deleted_since: nil, max_accounts_to_scrub: nil)
```

```suggestion
  # @param dry_run [Boolean] ...
  # @param scrub_accounts_deleted_since [Time] ...
  # @param max_accounts_to_scrub [Integer] ...
  def initialize(**options)
```